clang/lib/Lex/ModuleMap.cpp

```diff
@@ -984,7 +984,9 @@ static void inferFrameworkLink(Module *Mod) {
   assert(!Mod->isSubFramework() &&
          "Can only infer linking for top-level frameworks");
 
-  Mod->LinkLibraries.push_back(Module::LinkLibrary(Mod->Name,
+  StringRef FrameworkName(Mod->Name);
+  FrameworkName.consume_back("_Private");
+  Mod->LinkLibraries.push_back(Module::LinkLibrary(FrameworkName.str(),
```

Comment:
It would be better to avoid the allocation if the module name does not end with `"_Private"`.

Reply:
There is already an implicit allocation, because you need to pass a string to LinkLibrary.

Reply:
Ah, good point.